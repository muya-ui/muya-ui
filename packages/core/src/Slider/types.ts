import { ITriggerProps } from '../Trigger/types';
import { ICustomStyleBaseProps } from '../types';
import { Omit } from '../types/index';

/**
 * @styles startCircle 第一个滑块
 * @styles endCircle 第2个滑块
 * @styles markPoint 刻度在轨道上的点
 * @styles markPointInclude 刻度在轨道上的点，同时被区间包含
 * @styles marksRow 刻度的行
 * @styles markLabel 刻度的Label
 * @styles track 轨道
 * @styles range 激活区间
 */
export type ISliderStyleKeys =
  | 'startCircle'
  | 'endCircle'
  | 'markPoint'
  | 'markPointInclude'
  | 'marksRow'
  | 'markLabel'
  | 'track'
  | 'range';

export interface ISliderMarkItem {
  /** mark 容器的样式 */
  style?: React.CSSProperties;
  /** 标签本身 */
  label: React.ReactNode;
}

export type ISliderMark = React.ReactNode | ISliderMarkItem;

export interface ISliderMarks {
  /** key 必须在 min 和 max 之间的数值 */
  [key: number]: ISliderMark;
}

export interface ISliderBaseProps extends ICustomStyleBaseProps<ISliderStyleKeys> {
  /**
   * 最小值
   * @default 0
   */
  min?: number;
  /**
   * 最大值
   * @default 100
   */
  max?: number;
  /** 步进值 */
  step?: number;
  /** 是否禁用 */
  disabled?: boolean;
  /** 是否显示提示 */
  tooltipVisible?: boolean;
  /** 提示的位置 */
  tooltipPlacement?: ITriggerProps['placement'];
  /** 是否垂直 */
  vertical?: boolean;
  /**
   * 刻度标记，key 的类型必须为 number 且取值在闭区间 [min, max] 内，每个标签可以单独设置样式
   */
  marks?: ISliderMarks;
  /** 是否只能拖到刻度上 */
  marksOnly?: boolean;
  /** 不显示刻度的标签 */
  markLabelDisabled?: boolean;
  /**
   * 自动吸附到刻度上的距离，注意这里指的数值[min ~ max]，而不是UI上的距离
   * @default 1
   */
  snapDistance?: number;
  /**
   * 鼠标移动的节流设置
   * @default 50
   */
  mousemoveThrottleDelay?: number;
  /**
   * 数值有效小数位数
   * @default 2
   */
  fractionDigits?: number;
  /**
   * Slider 会把当前值传给 tipFormatter，并在 Tooltip 中显示 tipFormatter 的返回值，若为 null，则隐藏 Tooltip。
   * @default (value: number) => string
   * */
  tipFormatter?: (value: number) => React.ReactNode;
}

export interface ISliderPureProps extends ISliderBaseProps {
  /** Slider 的值 */
  value?: number;
  /** Slider 的默认值 */
  defaultValue?: number;
  /** 值变化事件 */
  onChange?: (value: number) => void;
  /** onmouseup 触发 */
  onAfterChange?: (value: number) => void;
}

export interface IRangeSliderPureProps extends ISliderBaseProps {
  /** RangeSlider 的值 */
  value?: number[];
  /** RangeSlider 的默认值 */
  defaultValue?: number[];
  onChange?: (value: [number, number]) => void;
  onAfterChange?: (value: [number, number]) => void;
}

type IDOMBase = Omit<React.HTMLAttributes<HTMLDivElement>, 'onChange' | 'defaultValue'>;

export type ISliderProps = ISliderPureProps & IDOMBase;
export type IRangeSliderProps = IRangeSliderPureProps & IDOMBase;
