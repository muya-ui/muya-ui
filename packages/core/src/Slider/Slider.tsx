import React from 'react';

import memoForwardRef from '../utils/memoForwardRef';
import BaseSlider from './BaseSlider';
import { IRangeSliderProps, ISliderProps } from './types';

export function useSlider(props: Pick<ISliderProps, 'onChange' | 'onAfterChange'>) {
  const { onChange, onAfterChange } = props;
  const handleChange = React.useCallback(
    (val: [number, number]) => {
      onChange && onChange(val[1]);
    },
    [onChange],
  );
  const handleAfterChange = React.useCallback(
    (val: [number, number]) => {
      onAfterChange && onAfterChange(val[1]);
    },
    [onAfterChange],
  );
  return {
    handleChange,
    handleAfterChange,
  };
}

const Slider = memoForwardRef<HTMLDivElement, ISliderProps>((props, ref) => {
  const { value, defaultValue, onChange, onAfterChange, min = 0, ...otherProps } = props;
  const restProps: IRangeSliderProps = {};
  if ('value' in props) {
    restProps['value'] = [min, props.value || min];
  }

  if ('defaultValue' in props) {
    restProps['defaultValue'] = [min, props.defaultValue || min];
  }
  const { handleChange, handleAfterChange } = useSlider(props);

  return (
    <BaseSlider
      ref={ref}
      hideStart
      onChange={handleChange}
      onAfterChange={handleAfterChange}
      min={min}
      {...otherProps}
      {...restProps}
    />
  );
});

export const RangeSlider = memoForwardRef<HTMLDivElement, IRangeSliderProps>((props, ref) => {
  return <BaseSlider {...props} ref={ref} />;
});

export default Slider;
