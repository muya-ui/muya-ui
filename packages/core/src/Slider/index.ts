export { default } from './Slider';
export { default as Slider } from './Slider';
export { RangeSlider } from './Slider';
export * from './types';
