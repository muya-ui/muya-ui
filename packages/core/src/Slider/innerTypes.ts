import { IRangeSliderProps } from './types';

export interface IBaseSliderProps extends IRangeSliderProps {
  hideStart?: boolean;
}
