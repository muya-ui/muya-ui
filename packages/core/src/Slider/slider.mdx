---
name: Slider 滑块
menu: 组件
route: /components/slider
autoDemos: true
parent: 数据录入
---

# Slider 滑块

滑动型输入器，展示当前值和可选范围。

## 使用场景

滑块可以让我们通过在连续或间断的区间内滑动锚点来选择一个合适的数值。区间最小值放在左边，对应的，最大值放在右边。滑块可以在滑动条的左右两端设定图标来反映数值的强度。这种交互特性使得它在设置诸如音量、亮度、色彩饱和度等需要反映强度等级的选项时成为一种极好的选择。

## 组件

- `Slider`: 滑块。
- `RangeSlider`: 区间滑块。

{Demos}

### API

#### ISliderProps

<Props propsName="ISliderProps" />

#### IRangeSliderProps

<Props propsName="IRangeSliderProps" />

#### ISliderMarks

刻度配置，可以是 `React.ReactNode` ，也可以是：

<Props propsName="ISliderMarkItem" />

### Styles

#### ISliderStyleKeys

<Styles typeName="ISliderStyleKeys" componentName="slider" />
