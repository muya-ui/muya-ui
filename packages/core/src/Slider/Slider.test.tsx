import React from 'react';
import renderer from 'react-test-renderer';
import sinon from 'sinon';

import { act, renderHook } from '@testing-library/react-hooks';

import Slider, { RangeSlider, useSlider } from './Slider';

test('测试 Slider', () => {
  const tree = renderer
    .create(
      <>
        <Slider />
        <Slider defaultValue={10} />
        <Slider value={10} />
        <Slider step={10} />
        <Slider step={10} markLabelDisabled />
      </>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('useSlider', () => {
  const onChange = sinon.spy();
  const onAfterChange = sinon.spy();
  const { result } = renderHook(() =>
    useSlider({
      onChange,
      onAfterChange,
    }),
  );
  result.current.handleChange([0, 10]);
  result.current.handleAfterChange([0, 10]);

  expect(() => {
    sinon.assert.calledWith(onChange, 10);
    sinon.assert.calledWith(onAfterChange, 10);
  }).not.toThrow();
});

test('测试 RangeSlider', () => {
  const tree = renderer
    .create(
      <>
        <RangeSlider />
        <RangeSlider defaultValue={[10, 12]} />
        <RangeSlider defaultValue={[18, 12]} />
        <RangeSlider value={[10, 20]} />
        <RangeSlider step={10} />
        <RangeSlider step={10} markLabelDisabled />
      </>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
