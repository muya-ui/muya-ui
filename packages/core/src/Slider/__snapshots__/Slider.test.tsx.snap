// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 RangeSlider 1`] = `
Array [
  .c0 {
  position: relative;
  box-sizing: border-box;
  cursor: pointer;
  padding: 4px 0;
}

.c0:hover .c1 {
  background-color: rgba(242,242,242,1);
}

.c4 {
  position: absolute;
  cursor: pointer;
  border-radius: 999rem;
  background-color: rgba(26,122,248,1);
  z-index: 1;
  height: 100%;
}

.c3 {
  position: absolute;
  z-index: 3;
  box-sizing: content-box;
  cursor: pointer;
  background-color: rgba(255,255,255,1);
  width: 12px;
  height: 12px;
  border-radius: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  top: 50%;
}

.c3::before {
  content: '';
  height: 100%;
  width: 100%;
  display: block;
  box-sizing: border-box;
  border-radius: 50%;
  background-color: rgba(255,255,255,1);
  border: 1px solid rgba(26,122,248,1);
}

.c3:hover {
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
  width: 16px;
  height: 16px;
  box-shadow: none;
}

.c3:hover::before {
  border-color: rgba(26,122,248,1);
}

.c2 {
  position: relative;
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  background-color: rgba(242,242,242,1);
  border-radius: 999rem;
  width: 100%;
  height: 6px;
}

<div
    className="c0"
  >
    <div
      className="c1 c2 muya-slider-track"
      onClick={[Function]}
    >
      <div
        className="c3 muya-slider-start-circle"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "left": "0%",
          }
        }
      />
      <div
        className="c3 muya-slider-end-circle"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "left": "0%",
          }
        }
      />
      <div
        className="c4 muya-slider-range"
        style={
          Object {
            "left": "0%",
          }
        }
      />
    </div>
  </div>,
  .c0 {
  position: relative;
  box-sizing: border-box;
  cursor: pointer;
  padding: 4px 0;
}

.c0:hover .c1 {
  background-color: rgba(242,242,242,1);
}

.c4 {
  position: absolute;
  cursor: pointer;
  border-radius: 999rem;
  background-color: rgba(26,122,248,1);
  z-index: 1;
  height: 100%;
}

.c3 {
  position: absolute;
  z-index: 3;
  box-sizing: content-box;
  cursor: pointer;
  background-color: rgba(255,255,255,1);
  width: 12px;
  height: 12px;
  border-radius: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  top: 50%;
}

.c3::before {
  content: '';
  height: 100%;
  width: 100%;
  display: block;
  box-sizing: border-box;
  border-radius: 50%;
  background-color: rgba(255,255,255,1);
  border: 1px solid rgba(26,122,248,1);
}

.c3:hover {
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
  width: 16px;
  height: 16px;
  box-shadow: none;
}

.c3:hover::before {
  border-color: rgba(26,122,248,1);
}

.c2 {
  position: relative;
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  background-color: rgba(242,242,242,1);
  border-radius: 999rem;
  width: 100%;
  height: 6px;
}

<div
    className="c0"
  >
    <div
      className="c1 c2 muya-slider-track"
      onClick={[Function]}
    >
      <div
        className="c3 muya-slider-start-circle"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "left": "10%",
          }
        }
      />
      <div
        className="c3 muya-slider-end-circle"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "left": "12%",
          }
        }
      />
      <div
        className="c4 muya-slider-range"
        style={
          Object {
            "left": "10%",
            "width": "2%",
          }
        }
      />
    </div>
  </div>,
  .c0 {
  position: relative;
  box-sizing: border-box;
  cursor: pointer;
  padding: 4px 0;
}

.c0:hover .c1 {
  background-color: rgba(242,242,242,1);
}

.c4 {
  position: absolute;
  cursor: pointer;
  border-radius: 999rem;
  background-color: rgba(26,122,248,1);
  z-index: 1;
  height: 100%;
}

.c3 {
  position: absolute;
  z-index: 3;
  box-sizing: content-box;
  cursor: pointer;
  background-color: rgba(255,255,255,1);
  width: 12px;
  height: 12px;
  border-radius: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  top: 50%;
}

.c3::before {
  content: '';
  height: 100%;
  width: 100%;
  display: block;
  box-sizing: border-box;
  border-radius: 50%;
  background-color: rgba(255,255,255,1);
  border: 1px solid rgba(26,122,248,1);
}

.c3:hover {
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
  width: 16px;
  height: 16px;
  box-shadow: none;
}

.c3:hover::before {
  border-color: rgba(26,122,248,1);
}

.c2 {
  position: relative;
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  background-color: rgba(242,242,242,1);
  border-radius: 999rem;
  width: 100%;
  height: 6px;
}

<div
    className="c0"
  >
    <div
      className="c1 c2 muya-slider-track"
      onClick={[Function]}
    >
      <div
        className="c3 muya-slider-start-circle"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "left": "12%",
          }
        }
      />
      <div
        className="c3 muya-slider-end-circle"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "left": "18%",
          }
        }
      />
      <div
        className="c4 muya-slider-range"
        style={
          Object {
            "left": "12%",
            "width": "6%",
          }
        }
      />
    </div>
  </div>,
  .c0 {
  position: relative;
  box-sizing: border-box;
  cursor: pointer;
  padding: 4px 0;
}

.c0:hover .c1 {
  background-color: rgba(242,242,242,1);
}

.c4 {
  position: absolute;
  cursor: pointer;
  border-radius: 999rem;
  background-color: rgba(26,122,248,1);
  z-index: 1;
  height: 100%;
}

.c3 {
  position: absolute;
  z-index: 3;
  box-sizing: content-box;
  cursor: pointer;
  background-color: rgba(255,255,255,1);
  width: 12px;
  height: 12px;
  border-radius: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  top: 50%;
}

.c3::before {
  content: '';
  height: 100%;
  width: 100%;
  display: block;
  box-sizing: border-box;
  border-radius: 50%;
  background-color: rgba(255,255,255,1);
  border: 1px solid rgba(26,122,248,1);
}

.c3:hover {
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
  width: 16px;
  height: 16px;
  box-shadow: none;
}

.c3:hover::before {
  border-color: rgba(26,122,248,1);
}

.c2 {
  position: relative;
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  background-color: rgba(242,242,242,1);
  border-radius: 999rem;
  width: 100%;
  height: 6px;
}

<div
    className="c0"
  >
    <div
      className="c1 c2 muya-slider-track"
      onClick={[Function]}
    >
      <div
        className="c3 muya-slider-start-circle"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "left": "10%",
          }
        }
      />
      <div
        className="c3 muya-slider-end-circle"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "left": "20%",
          }
        }
      />
      <div
        className="c4 muya-slider-range"
        style={
          Object {
            "left": "10%",
            "width": "10%",
          }
        }
      />
    </div>
  </div>,
  .c0 {
  position: relative;
  box-sizing: border-box;
  cursor: pointer;
  padding: 4px 0;
}

.c0:hover .c1 {
  background-color: rgba(242,242,242,1);
}

.c4 {
  position: absolute;
  cursor: pointer;
  border-radius: 999rem;
  background-color: rgba(26,122,248,1);
  z-index: 1;
  height: 100%;
}

.c3 {
  position: absolute;
  z-index: 3;
  box-sizing: content-box;
  cursor: pointer;
  background-color: rgba(255,255,255,1);
  width: 12px;
  height: 12px;
  border-radius: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  top: 50%;
}

.c3::before {
  content: '';
  height: 100%;
  width: 100%;
  display: block;
  box-sizing: border-box;
  border-radius: 50%;
  background-color: rgba(255,255,255,1);
  border: 1px solid rgba(26,122,248,1);
}

.c3:hover {
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
  width: 16px;
  height: 16px;
  box-shadow: none;
}

.c3:hover::before {
  border-color: rgba(26,122,248,1);
}

.c2 {
  position: relative;
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  background-color: rgba(242,242,242,1);
  border-radius: 999rem;
  width: 100%;
  height: 6px;
}

<div
    className="c0"
  >
    <div
      className="c1 c2 muya-slider-track"
      onClick={[Function]}
    >
      <div
        className="c3 muya-slider-start-circle"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "left": "0%",
          }
        }
      />
      <div
        className="c3 muya-slider-end-circle"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "left": "0%",
          }
        }
      />
      <div
        className="c4 muya-slider-range"
        style={
          Object {
            "left": "0%",
          }
        }
      />
    </div>
  </div>,
  .c0 {
  position: relative;
  box-sizing: border-box;
  cursor: pointer;
  padding: 4px 0;
}

.c0:hover .c1 {
  background-color: rgba(242,242,242,1);
}

.c4 {
  position: absolute;
  cursor: pointer;
  border-radius: 999rem;
  background-color: rgba(26,122,248,1);
  z-index: 1;
  height: 100%;
}

.c3 {
  position: absolute;
  z-index: 3;
  box-sizing: content-box;
  cursor: pointer;
  background-color: rgba(255,255,255,1);
  width: 12px;
  height: 12px;
  border-radius: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  top: 50%;
}

.c3::before {
  content: '';
  height: 100%;
  width: 100%;
  display: block;
  box-sizing: border-box;
  border-radius: 50%;
  background-color: rgba(255,255,255,1);
  border: 1px solid rgba(26,122,248,1);
}

.c3:hover {
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
  width: 16px;
  height: 16px;
  box-shadow: none;
}

.c3:hover::before {
  border-color: rgba(26,122,248,1);
}

.c2 {
  position: relative;
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  background-color: rgba(242,242,242,1);
  border-radius: 999rem;
  width: 100%;
  height: 6px;
}

<div
    className="c0"
  >
    <div
      className="c1 c2 muya-slider-track"
      onClick={[Function]}
    >
      <div
        className="c3 muya-slider-start-circle"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "left": "0%",
          }
        }
      />
      <div
        className="c3 muya-slider-end-circle"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "left": "0%",
          }
        }
      />
      <div
        className="c4 muya-slider-range"
        style={
          Object {
            "left": "0%",
          }
        }
      />
    </div>
  </div>,
]
`;

exports[`测试 Slider 1`] = `
Array [
  .c0 {
  position: relative;
  box-sizing: border-box;
  cursor: pointer;
  padding: 4px 0;
}

.c0:hover .c1 {
  background-color: rgba(242,242,242,1);
}

.c4 {
  position: absolute;
  cursor: pointer;
  border-radius: 999rem;
  background-color: rgba(26,122,248,1);
  z-index: 1;
  height: 100%;
}

.c3 {
  position: absolute;
  z-index: 3;
  box-sizing: content-box;
  cursor: pointer;
  background-color: rgba(255,255,255,1);
  width: 12px;
  height: 12px;
  border-radius: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  top: 50%;
}

.c3::before {
  content: '';
  height: 100%;
  width: 100%;
  display: block;
  box-sizing: border-box;
  border-radius: 50%;
  background-color: rgba(255,255,255,1);
  border: 1px solid rgba(26,122,248,1);
}

.c3:hover {
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
  width: 16px;
  height: 16px;
  box-shadow: none;
}

.c3:hover::before {
  border-color: rgba(26,122,248,1);
}

.c2 {
  position: relative;
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  background-color: rgba(242,242,242,1);
  border-radius: 999rem;
  width: 100%;
  height: 6px;
}

<div
    className="c0"
  >
    <div
      className="c1 c2 muya-slider-track"
      onClick={[Function]}
    >
      <div
        className="c3 muya-slider-end-circle"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "left": "0%",
          }
        }
      />
      <div
        className="c4 muya-slider-range"
        style={
          Object {
            "left": "0%",
          }
        }
      />
    </div>
  </div>,
  .c0 {
  position: relative;
  box-sizing: border-box;
  cursor: pointer;
  padding: 4px 0;
}

.c0:hover .c1 {
  background-color: rgba(242,242,242,1);
}

.c4 {
  position: absolute;
  cursor: pointer;
  border-radius: 999rem;
  background-color: rgba(26,122,248,1);
  z-index: 1;
  height: 100%;
}

.c3 {
  position: absolute;
  z-index: 3;
  box-sizing: content-box;
  cursor: pointer;
  background-color: rgba(255,255,255,1);
  width: 12px;
  height: 12px;
  border-radius: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  top: 50%;
}

.c3::before {
  content: '';
  height: 100%;
  width: 100%;
  display: block;
  box-sizing: border-box;
  border-radius: 50%;
  background-color: rgba(255,255,255,1);
  border: 1px solid rgba(26,122,248,1);
}

.c3:hover {
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
  width: 16px;
  height: 16px;
  box-shadow: none;
}

.c3:hover::before {
  border-color: rgba(26,122,248,1);
}

.c2 {
  position: relative;
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  background-color: rgba(242,242,242,1);
  border-radius: 999rem;
  width: 100%;
  height: 6px;
}

<div
    className="c0"
  >
    <div
      className="c1 c2 muya-slider-track"
      onClick={[Function]}
    >
      <div
        className="c3 muya-slider-end-circle"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "left": "10%",
          }
        }
      />
      <div
        className="c4 muya-slider-range"
        style={
          Object {
            "left": "0%",
            "width": "10%",
          }
        }
      />
    </div>
  </div>,
  .c0 {
  position: relative;
  box-sizing: border-box;
  cursor: pointer;
  padding: 4px 0;
}

.c0:hover .c1 {
  background-color: rgba(242,242,242,1);
}

.c4 {
  position: absolute;
  cursor: pointer;
  border-radius: 999rem;
  background-color: rgba(26,122,248,1);
  z-index: 1;
  height: 100%;
}

.c3 {
  position: absolute;
  z-index: 3;
  box-sizing: content-box;
  cursor: pointer;
  background-color: rgba(255,255,255,1);
  width: 12px;
  height: 12px;
  border-radius: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  top: 50%;
}

.c3::before {
  content: '';
  height: 100%;
  width: 100%;
  display: block;
  box-sizing: border-box;
  border-radius: 50%;
  background-color: rgba(255,255,255,1);
  border: 1px solid rgba(26,122,248,1);
}

.c3:hover {
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
  width: 16px;
  height: 16px;
  box-shadow: none;
}

.c3:hover::before {
  border-color: rgba(26,122,248,1);
}

.c2 {
  position: relative;
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  background-color: rgba(242,242,242,1);
  border-radius: 999rem;
  width: 100%;
  height: 6px;
}

<div
    className="c0"
  >
    <div
      className="c1 c2 muya-slider-track"
      onClick={[Function]}
    >
      <div
        className="c3 muya-slider-end-circle"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "left": "10%",
          }
        }
      />
      <div
        className="c4 muya-slider-range"
        style={
          Object {
            "left": "0%",
            "width": "10%",
          }
        }
      />
    </div>
  </div>,
  .c0 {
  position: relative;
  box-sizing: border-box;
  cursor: pointer;
  padding: 4px 0;
}

.c0:hover .c1 {
  background-color: rgba(242,242,242,1);
}

.c4 {
  position: absolute;
  cursor: pointer;
  border-radius: 999rem;
  background-color: rgba(26,122,248,1);
  z-index: 1;
  height: 100%;
}

.c3 {
  position: absolute;
  z-index: 3;
  box-sizing: content-box;
  cursor: pointer;
  background-color: rgba(255,255,255,1);
  width: 12px;
  height: 12px;
  border-radius: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  top: 50%;
}

.c3::before {
  content: '';
  height: 100%;
  width: 100%;
  display: block;
  box-sizing: border-box;
  border-radius: 50%;
  background-color: rgba(255,255,255,1);
  border: 1px solid rgba(26,122,248,1);
}

.c3:hover {
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
  width: 16px;
  height: 16px;
  box-shadow: none;
}

.c3:hover::before {
  border-color: rgba(26,122,248,1);
}

.c2 {
  position: relative;
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  background-color: rgba(242,242,242,1);
  border-radius: 999rem;
  width: 100%;
  height: 6px;
}

<div
    className="c0"
  >
    <div
      className="c1 c2 muya-slider-track"
      onClick={[Function]}
    >
      <div
        className="c3 muya-slider-end-circle"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "left": "0%",
          }
        }
      />
      <div
        className="c4 muya-slider-range"
        style={
          Object {
            "left": "0%",
          }
        }
      />
    </div>
  </div>,
  .c0 {
  position: relative;
  box-sizing: border-box;
  cursor: pointer;
  padding: 4px 0;
}

.c0:hover .c1 {
  background-color: rgba(242,242,242,1);
}

.c4 {
  position: absolute;
  cursor: pointer;
  border-radius: 999rem;
  background-color: rgba(26,122,248,1);
  z-index: 1;
  height: 100%;
}

.c3 {
  position: absolute;
  z-index: 3;
  box-sizing: content-box;
  cursor: pointer;
  background-color: rgba(255,255,255,1);
  width: 12px;
  height: 12px;
  border-radius: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  top: 50%;
}

.c3::before {
  content: '';
  height: 100%;
  width: 100%;
  display: block;
  box-sizing: border-box;
  border-radius: 50%;
  background-color: rgba(255,255,255,1);
  border: 1px solid rgba(26,122,248,1);
}

.c3:hover {
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
  width: 16px;
  height: 16px;
  box-shadow: none;
}

.c3:hover::before {
  border-color: rgba(26,122,248,1);
}

.c2 {
  position: relative;
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  background-color: rgba(242,242,242,1);
  border-radius: 999rem;
  width: 100%;
  height: 6px;
}

<div
    className="c0"
  >
    <div
      className="c1 c2 muya-slider-track"
      onClick={[Function]}
    >
      <div
        className="c3 muya-slider-end-circle"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "left": "0%",
          }
        }
      />
      <div
        className="c4 muya-slider-range"
        style={
          Object {
            "left": "0%",
          }
        }
      />
    </div>
  </div>,
]
`;
